google_guest_agent/non_windows_accounts_integ_test.go

```diff
@@ -1,12 +1,76 @@
+//go:build integration
 // +build integration
 
 package main
 
 import (
+	"fmt"
+	"os"
 	"os/exec"
+	"strings"
 	"testing"
 )
 
+const (
+	testUser       = "integration-test-user"
+	expectedGroups = []string{"adm", "dip", "docker", "lxd", "plugdev", "video", "google-sudoers"}
+)
+
+func TestCreateGoogleUser(t *testing.T) {
+	if exist, err := userExists(testUser); exist == true {
+		t.Errorf("test user should not exist")
+	}
+	if err := createGoogleUser(testUser); err != nil {
+		t.Errorf("failed creating test user")
+	}
+	if exist, err := userExists(testUser); exist != true || err != nil {
+		t.Errorf("tesr user should exist")
+	}
+	cmd := exec.Command("groups", testUser)
+	ret := runCmdOutput(cmd)
+	if ret.ExitCode() != 0 {
+		t.Errorf("test user should be added to group")
+	}
+	groups := strings.Split(strings.TrimSpace(strings.Split(ret.Stdout(), ":")[1]), " ")
+	for _, group := range groups {
+		if !contains(group, expectedGroups) {
+			t.Errorf("test user has been added to an unexpected group %s", group)
+		}
+	}
+	for _, expected := range expectedGroups {
+		if !contains(expected, groups) {
+			t.Errorf("test user has not been added to group %s", expected)
+		}
+	}
+	if _, err := os.Stat(fmt.Sprintf("/home/%s", testUser)); err != nil {
+		t.Errorf("test user home directory is not exist")
+	}
+	if err := createGoogleUser(testUser); err == nil {
+		t.Errorf("user should exist and return error but not")
+	}
+}
+
+func TestRemoveGoogleUser(t *testing.T) {
+	if exist, err := userExists(testUser); exist == true {
+		t.Errorf("test user should not exist")
+	}
+	if err := createGoogleUser(testUser); err != nil {
+		t.Errorf("failed creating test user")
+	}
+	if err := removeGoogleUser(testUser); err != nil {
+		t.Errorf("failed when remove google user")
+	}
+	if exist, err := userExists(testUser); exist == true {
+		t.Errorf("test user should not exist")
+	}
+	if _, err := os.Stat(fmt.Sprintf("/home/%s", testUser)); err == nil {
+		t.Errorf("test user home directory should not exist")
+	}
```

Comment:
we only remove the home directory if configured to do so. for this simple test, just check whether we were supposed to. but we need to add a test that explicitly sets this config option and confirms it took effect. that could be a different unit test, or just turn this test into one that deals with multiple cases
```suggestion
	if config.Section("Accounts").Key("deprovision_remove").MustBool(false) {
		if _, err := os.Stat(fmt.Sprintf("/home/%s", testUser)); err == nil {
			t.Errorf("test user home directory should not exist")
		}
	}
```